client/sites-dashboard/components/sites-table-row.tsx

```diff
@@ -130,13 +131,19 @@ export default memo( function SitesTableRow( { site }: SiteTableRowProps ) {
 			</Column>
 			<Column mobileHidden>
 				<SitePlan>
-					{ isNotAtomicJetpack( site ) && (
+					{ isNotAtomicJetpack( site ) && ! site.plan?.expired && (
 						<SitePlanIcon>
 							<JetpackLogo size={ 16 } />
 						</SitePlanIcon>
 					) }
-
-					{ site.plan?.product_name_short }
+					{ site.plan?.expired ? (
+						<PlanRenewNag
+							plan={ site.plan }
+							checkoutUrl={ `/checkout/${ site.slug }/${ site.plan?.product_slug }` }
+						/>
+					) : (
+						<>{ site.plan?.product_name_short }</>
+					) }
```

Comment:
I suggest removing the unnecessary fragment.

```suggestion
					) : (
						site.plan?.product_name_short
					) }
```